styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -422,6 +449,15 @@ private static void startScheduler(Scheduler scheduler, ScheduledExecutorService
         TimeUnit.SECONDS);
   }
 
+  private static void startSubmissionRateLimiterUpdate(Storage storage, RateLimiter rateLimiter,
+      ScheduledExecutorService exec) {
+    exec.scheduleAtFixedRate(
+        guard(() -> updateSubmissionRateLimiter(storage, rateLimiter)),
+        SUBMISSION_RATE_LIMITER_UPDATE_TICK_INTERVAL_SECONDS,
```

Comment:
Delay in scheduling is arguably not a big deal, but I -1 astonishingly starting a service with default configuration when the user has configured something else.

Specifying 0 would not be perfect anyway in the current setup I guess (racy) but would get +1, perfect being the enemy of good...